File: SocketComm.Examples/SendValue/IntServer.java
import java.nio.ByteBuffer;

public class IntServer {
    public static void main(String[] args) throws Exception {
        FPipe p = new FPipe("/tmp/intvalue.pipe");
        p.ExpectCmd(Cmd.Ready);
        int i = 2;
        while (i < Integer.MAX_VALUE && i > 1)
        {
            System.out.printf("Sending %d\n", i);
            p.SendValue(i, (val) -> ByteBuffer.allocate(4).putInt(val));
            p.ExpectCmd(Cmd.Ready);
            p.WriteCmd(Cmd.Go);
            p.ExpectCmd(Cmd.Done);
            p.ExpectCmd(Cmd.Receive);
            i = p.ReceiveValue(ByteBuffer::getInt);
            System.out.printf("Received %d\n", i);
        }
        p.WriteCmd(Cmd.Exit);
        p.close();
    }
}

File: SocketComm.Examples/SendValue/IntServer.cs
using static SocketComm.Cmd;
namespace SocketComm.Examples.SendValue;

public class IntServer
{
    public static void Main()
    {
        FPipe p = new FPipe("/tmp/intvalue.pipe");
        p.Connect();
        try
        {
            p.ExpectCmd(Ready);
            int i = 2;
            while (i is < Int32.MaxValue and > 1)
            {
                Console.WriteLine($"Sending {i}");
                p.SendValue(i, SimpleConversion.NumberToBytes);
                p.ExpectCmd(Ready);
                p.WriteCmd(Go);
                p.ExpectCmd(Done);
                p.ExpectCmd(Receive);
                i = p.ReceiveValue(SimpleConversion.BytesToNumber<int>);
                Console.WriteLine($"Received {i}");
            }
        }
        catch (Exception e)
        {
            Console.Error.WriteLine("An error occured with the pipe: " + e);
        }
        finally
        {
            p.WriteCmd(Exit);
            p.Dispose();
        }
        
    }
}